extensions/kafka/src/main/java/io/deephaven/kafka/publish/PublishToKafka.java

```diff
@@ -249,16 +249,13 @@ private PublishListener(
 
         @Override
         public void onUpdate(TableUpdate upstream) {
-            if (keySerializer != null || isStream) {
```

Comment:
I'm assuming `keySerializer != null` was just a bad check here too?

Reply:
No. The theory was that if you had a key serializer, we expected you to be downstream of an aggregation, in which case there would be no shifts.